### PR TITLE
fix: partial liquidation DOS

### PR DESCRIPTION
**Pull request by 0xJabberwock**
Closes HAI-312

### CODE REVIEW

File: src/contracts/LiquidationEngine.sol [hunk 1]
```diff
@@ -174,11 +174,6 @@ contract LiquidationEngine is
         if (_limitAdjustedDebt == 0) revert LiqEng_NullAuction();
 
         if (_collateralToSell == 0) revert LiqEng_NullCollateralToSell();
-
-        if (
-          _limitAdjustedDebt != _safeData.generatedDebt
-            && (_safeData.generatedDebt - _limitAdjustedDebt) * _safeEngCData.accumulatedRate < _debtFloor
-        ) revert LiqEng_DustySAFE();
```
Comment (wei3erHase) [inline src/contracts/LiquidationEngine.sol:181]: hmmm i'm not sure this is the expected behaviour, letme think a bit...

File: src/contracts/LiquidationEngine.sol [hunk 2]
```diff
@@ -175,10 +175,16 @@ contract LiquidationEngine is
...
+        if (currentOnAuctionSystemCoins + _limitAdjustedDebt > _params.onAuctionSystemCoinLimit) {
+          revert LiqEng_ExceededCapacity();
+        }
+
         if (
           _limitAdjustedDebt != _safeData.generatedDebt
             && (_safeData.generatedDebt - _limitAdjustedDebt) * _safeEngCData.accumulatedRate < _debtFloor
-        ) revert LiqEng_DustySAFE();
+        ) {
+          _limitAdjustedDebt = _safeData.generatedDebt;
```
Comment (0xJabberwock) [inline src/contracts/LiquidationEngine.sol]: Previous variables such as `_collateralToSell` and `_amountToRaise` are computed based on the first `_limitAdjustedDebt` value.

Comment (wei3erHase) [reply src/contracts/LiquidationEngine.sol]: hmm we should need to recompute them? letme see.

Comment (wei3erHase) [reply src/contracts/LiquidationEngine.sol]: ok, so, this is not a 'health check' anymore, let's bring this logic a few lines up, so the other variables are updated accordingly 🙏

**Review by wei3erHase — CHANGES_REQUESTED** [review]
let's remove it from the "safety checks" scope, and update the pointed out variables 🙏

File: test/unit/LiquidationEngine.t.sol [hunk 3]
```diff
@@ -884,6 +911,16 @@ contract Unit_LiquidationEngine_LiquidateSafe is Base {
 
   function _assumeHappyPathFullLiquidation(Liquidation memory _liquidation) internal pure {
```
Comment (0xJabberwock) [inline test/unit/LiquidationEngine.t.sol:912]: Fuzzer fails to handle the `happyPathFullLiquidation`. 😭

File: src/contracts/LiquidationEngine.sol [hunk 4]
```diff
@@ -175,10 +180,9 @@ contract LiquidationEngine is
 
         if (_collateralToSell == 0) revert LiqEng_NullCollateralToSell();
 
-        if (
-          _limitAdjustedDebt != _safeData.generatedDebt
-            && (_safeData.generatedDebt - _limitAdjustedDebt) * _safeEngCData.accumulatedRate < _debtFloor
-        ) revert LiqEng_DustySAFE();
+        if (currentOnAuctionSystemCoins + _limitAdjustedDebt > _params.onAuctionSystemCoinLimit) {
+          revert LiqEng_ExceededCapacity();
+        }
```
Comment (wei3erHase) [inline src/contracts/LiquidationEngine.sol:185]: this is the only line where we should consider the `onAuctionSystemCoinLimit`, please remove it from the 1st `limitAdjustedDebt` calculation.
if the limit is reached, we don't cap the amount to liquidate, we just revert 🙏

**Comment by wei3erHase** [comment]
closed in favour of #25

**Pull request closed without merging**